docs/Glossary.md

```diff
@@ -0,0 +1,141 @@
+
+## Action
+
+An _action_ is an object that represents an intention to change the state. Actions are the only way to get data into the store. Any data, whether from UI events, network callbacks, system events, or other sources such as WebSockets needs to eventually be dispatched as actions.
+
+ReduxKotlin differs a bit from JS Redux in that since we have a statically typed language, we can use the type as indication of the action to be performed. Typically this is done with a `when` statement:
+
+```kotlin
+	when (action) {
+	   is MyAction -> //handle MyAction
+```
+
+In JS redux, a `type` field is used to indicate the type of action performed.  This approach can be used with ReduxKotlin, however type safety and IDE benefits will be lost.
+
+The structure of an action object is really up to you.  Data classes are a good choice, since `toString()` is impemented and can be useful for logging. Data inside the action is used to update the state inside reducers.  For example, an API request result may be inside an Action.
+
+See also [async action](#async-action) below.
+
+## Reducer
+
+```kotlin
+typealias Reducer<State> = (state: State, action: Any) -> State
+```
+
+A _reducer_ (also called a _reducing function_) is a function that accepts an accumulation and a value and returns a new accumulation. They are used to reduce a collection of values down to a single value.
+
+Reducers are not unique to Redux—they are a fundamental concept in functional programming. Even most non-functional languages, like Kotlin, have a built-in API for reducing. In Kotlin, reduce is a[`function on collections and many types`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/reduce.html).
+
+In Redux, the accumulated value is the state object, and the values being accumulated are actions. Reducers calculate a new state given the previous state and an action. They must be _pure functions_—functions that return the exact same output for given inputs. They should also be free of side-effects. This is what enables exciting features like hot reloading and time travel.
+
+Reducers are the most important concept in Redux.
+
+_Do not put API calls into reducers._
+
+## Dispatching Function
+
+```kotlin
+typealias Dispatcher = (Any) -> Any
+```
+
+A _dispatching function_ (or simply _dispatch function_) is a function that accepts an action or an [async action](#async-action); it then may or may not dispatch one or more actions to the store.
+
+We must distinguish between dispatching functions in general and the base [`dispatch`](api/Store.md#dispatchaction) function provided by the store instance without any middleware.
+
+The base dispatch function _always_ synchronously sends an action to the store's reducer, along with the previous state returned by the store, to calculate a new state. It expects actions to be plain objects ready to be consumed by the reducer.
```

Comment:
This might be a good idea to mention that updating state should be thread confined

Reply:
Yes, good idea.  Thinking there could also be a page just on threading and have a link here

Reply:
Sounds good 👌

Reply:
link added